src/files/file/File.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import prettyBytes from 'pretty-bytes'
+import Checkbox from '../../components/checkbox/Checkbox'
+import extToPic from './ext-to-pic'
+import Status from '../status/Status'
+import './File.css'
+
+const File = (props) => {
+  let {selected, name, type, speed, status, size, hash, onSelect, onNavigate, onCancel} = props
+
+  let className = 'File flex items-center bt pv2'
+
+  if (props.selected) {
+    className += ' selected'
+  }
+
+  if (status !== null) {
+    size = 'N/A'
+    status = <Status progress={status} cancel={onCancel} speed={speed} />
+  } else {
+    size = prettyBytes(size)
+  }
+
+  if (type === 'directory') {
+    size = ''
+  }
+
+  const select = (select) => {
+    onSelect(hash, select)
+  }
+
+  return (
+    <div className={className}>
+      <div className='pa2 w2'>
+        <Checkbox checked={selected} onChange={select} />
+      </div>
+      <div className='name flex items-center flex-grow-1 pa2 w-40'>
+        <div className='dib icon'>
+          {extToPic(props)}
```

Comment:
@hacdias with a little tweak this can be a JSX element. if we rename it FileIcon we can

```jsx
<FileIcon name={filename} /> 
```